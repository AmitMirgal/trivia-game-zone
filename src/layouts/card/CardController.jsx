import React from "react";
import styled from "styled-components";
import { PropTypes } from "prop-types";

const Controller = styled.div`
  display: flex;
  flex-direction: ${(props) => (props.row ? "row" : "column")};
  align-items: flex-start;
  padding: 12px;
  margin: 12px;
`;

const CardController = (props) => {
  const { children, alignRow } = props;
  return <Controller row={alignRow}>{children}</Controller>;
};

CardController.propTypes = {
  children: PropTypes.node,
  alignRow: PropTypes.bool,
};

export default CardController;
