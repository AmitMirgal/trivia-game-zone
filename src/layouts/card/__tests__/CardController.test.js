import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import CardController from "../CardController";

describe("<CardController/>", () => {
  let wrapper;
  const prepare = () => {
    wrapper = shallow(<CardController />);
  };

  it("renders CardController component", () => {
    // when
    prepare();

    // then
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("count number of props", () => {
    // when
    prepare();

    const propsSize = Object.keys(wrapper.props()).length;

    // then
    expect(propsSize).toBe(2);
  });
});
