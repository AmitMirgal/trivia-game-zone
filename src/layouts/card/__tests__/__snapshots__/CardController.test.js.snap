// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardController/> renders CardController component 1`] = `<styled.div />`;
